src/components/TextInput.tsx

```diff
@@ -1,63 +1,53 @@
-/**
- * @author Angela Guo <aguo921@aucklanduni.ac.nz>
- */
-
 "use client";
 
-import { useState } from "react";
+import type { InputHTMLAttributes } from "react";
 import { twMerge } from "tailwind-merge";
 
 import { cn } from "@/lib/utils";
 
-type TextInputProps = {
-  label?: string;
-  value: string;
+interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+  label: string;
   type: string;
-  isError: boolean;
-  backgroundColor?: string;
-  className?: string;
-  onChange: (value: string) => void;
-};
+  isError?: boolean;
+  OnChange?: (value: string) => void;
+  classNames?: string;
+}
 
-export const TextInput = (props: TextInputProps) => {
-  const [active, setActive] = useState(false);
+/**
+ * @param {React.Dispatch<SetStateAction<any>>} OnChange (e) => OnChange?(e.target.value)
+ */
 
+export const TextInput = ({
+  label,
+  type,
+  isError,
+  className,
+  OnChange,
+  ...props
+}: InputProps) => {
   return (
-    <div className={cn("relative h-11", props.className)}>
-      <h2
-        className={twMerge(
-          cn(
-            "absolute left-3 transition-all pointer-events-none z-10",
-            props.backgroundColor,
-          ),
-          active || props.value != ""
-            ? "top-[-0.75rem] px-2 text-sm  text-blue-500"
-            : "top-1.5 cursor-text text-gray-500",
-          props.isError &&
-            "absolute left-3 top-[-0.75rem] px-2 text-sm  text-red-500",
-        )}
-      >
-        {props.label}
-      </h2>
-
+    <div className={cn("relative h-11 ", className)}>
       <input
-        type={props.type}
-        defaultValue={props.value}
-        onChange={(e) => props.onChange(e.target.value)}
-        onFocus={() => setActive(true)}
-        onBlur={() => setActive(false)}
+        type={type}
+        placeholder=" "
         className={twMerge(
-          cn(
-            "w-full rounded p-2 border focus:border-2 outline-none h-full",
-            props.backgroundColor,
-          ),
-          active
-            ? "border-blue-500 focus:border-blue-500"
-            : "border-blue-400 focus:border-blue-400",
-          props.isError && "border-red-500 focus:border-red-500",
+          "peer w-full border-primary rounded p-2 border focus:border-2 outline-none h-full bg-background text-tertiary dark:text-white/70",
+          isError && "border-red-500 ",
         )}
+        {...props}
+        onChange={(e) => OnChange?.(e.target.value)}
         required
       />
+      <h2
+        className={cn(
+          "absolute -top-3 left-2 transition-all pointer-events-none select-none z-10 bg-background px-2 text-primary text-xs",
+          "peer-focus:-top-3 peer-focus:translate-y-[0%] peer-focus:text-primary peer-focus:text-xs",
```

Comment:
```suggestion
          "absolute -top-3 left-2 transition-all pointer-events-none select-none z-10 px-1 text-primary text-xs",
```

I think px-2 is too much space lol